alerts/charts/templates/tests/test-alerts-config.yaml

```
@@ -61,4 +48,18 @@ data:
         verify "there is 1 customresourcedefinition named 'thanosrulers'"
     }
     {{- end }}
+
+    {{- if and .Values.alerts.auth.autoGenerateCert.enabled .Values.alerts.alertmanager.ingress.enabled .Values.alerts.alertmanager.ingress.hosts (.Capabilities.APIVersions.Has "cert-manager.io/v1") }}
+    @test "Verify certificate creation and validation against the API endpoint of the alert manager" {
+        verify "there is 1 issuer named '{{ .Release.Namespace }}-monitoring-issuer'"
+        verify "there is 1 certificate named '{{ .Release.Namespace }}-prometheus-auth'"
+        verify "there is 1 secret named 'tls-{{ .Release.Namespace }}-prometheus-auth'"
+
+        url="https://{{ first .Values.alerts.alertmanager.ingress.hosts }}/-/healthy"
+
+        response=$(curl --cert /tls-assets/tls.crt --key /tls-assets/tls.key ${url})
+        [ "$response" == "OK" ]
```

Suggesting to remove not needed var which looks more native to me according to various examples in the documentation. At least if we're not checking for a particular return string. OK should yield return code `0`.

```suggestion
        run curl --cert /tls-assets/tls.crt --key /tls-assets/tls.key ${url}
        [ "$status" -eq "0" ]
```